libcontainer/cgroups/fs2/hugetlb.go

```diff
@@ -39,15 +59,23 @@ func statHugeTlb(dirPath string, stats *cgroups.Stats) error {
 	}
 	hugetlbStats := cgroups.HugetlbStats{}
 
+	supportsReservationAccounting := HasReservationAccountingSupport(dirPath)
+
 	for _, pagesize := range hugePageSizes {
-		usage := strings.Join([]string{"hugetlb", pagesize, "current"}, ".")
+		filenamePrefix := strings.Join([]string{"hugetlb", pagesize}, ".")
+
+		if supportsReservationAccounting {
+			filenamePrefix += ".rsvd"
+		}
+
+		usage := fmt.Sprintf("%s.current", filenamePrefix)
 		value, err := fscommon.GetCgroupParamUint(dirPath, usage)
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse hugetlb.%s.current file", pagesize)
```

Comment:
The error message from `GetCgroupParamUint` already contain file name, so you can return the error as-is, no need to wrap it.

Reply:
also the error now returns the wrong file name in case `supportsReservationAccounting` is set

Reply:
Fixing this should be done as a separate first patch I think.